erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -852,8 +855,11 @@ def make_stock_adjustment_entry(self, gl_entries, item, voucher_wise_stock_value
 	def make_tax_gl_entries(self, gl_entries):
 		# tax table gl entries
 		valuation_tax = {}
+		enable_discount_accounting = cint(frappe.db.get_single_value('Accounts Settings', 'enable_discount_accounting'))
```

Comment:
Instead of fetching `enable_discount_accounting` twice, you can make it a class property.

Reply:
Done.